tests/robustness/robustness_test/robustness_test.go

```diff
@@ -23,15 +25,67 @@ func TestManySmallFiles(t *testing.T) {
 		engine.MaxNumFilesPerWriteField: strconv.Itoa(numFiles),
 		engine.MinNumFilesPerWriteField: strconv.Itoa(numFiles),
 	}
+	var errs errgroup.Group
+	//b := &strings.Builder{}
+	log.Printf("Printing the length: %v", len(eng.Checker))
+	for i := range eng.Checker {
+		func(index int) {
+			errs.Go(func() error {
+				log.Printf("Printing the Engine: %v", index)
+				_, err := eng.ExecAction(engine.WriteRandomFilesActionKey, fileWriteOpts, index)
+				if err != nil {
+					return err
+				}
+
+				snapOut, err := eng.ExecAction(engine.SnapshotRootDirActionKey, nil, index)
+				if err != nil {
+					return err
+				}
+				_, err = eng.ExecAction(engine.RestoreSnapshotActionKey, snapOut, index)
+				return err
+
+			})
+		}(i)
+	}
 
-	_, err := eng.ExecAction(engine.WriteRandomFilesActionKey, fileWriteOpts)
-	testenv.AssertNoError(t, err)
-
-	snapOut, err := eng.ExecAction(engine.SnapshotRootDirActionKey, nil)
+	err := errs.Wait()
 	testenv.AssertNoError(t, err)
 
-	_, err = eng.ExecAction(engine.RestoreSnapshotActionKey, snapOut)
-	testenv.AssertNoError(t, err)
+	// for i := range eng.Checker {
```

Comment:
This was the code that I tried first to run every `ExecAction` func call as a go routine, but this seems to not complete indefinitely. Just kept this here fore reference.